test/e2e/csc_tests.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+const (
+	testResourceName string = "test-operators"
+)
+
+func runCSCWithInvalidTargetNamespace(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	f := test.Global
+	// Run tests
+	if err := cscWithInvalidTargetNamespace(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func cscWithInvalidTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Define variables to be used in test
+	targetNamespace := "nonexisting-namespace"
+	packages := "etcd,federationv2,mongodb,amq-streams,couchbase,descheduler,prometheus,svcat,clusterlogging,dynatrace,metering"
+	registryNamespace := "marketplace_e2e"
+
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create the operatorsource to download the manifests
+	testOperatorSource := createOperatorSource(testResourceName, namespace, nil, registryNamespace)
```

Comment:
For CatalogSourceConfig tests, if we are running these in serial, would it be better if we did some upfront setup like create one operatorsource beforehand, and then use the same one for all the tests? Rather than create all these resources each time for each test?

Reply:
This is certainly something I had considered. I plan to do a follow up PR that refactors existing tests which would include this work.